src/graphql/root/mutation/ln-noamount-invoice-payment-send.ts

```diff
@@ -17,23 +18,42 @@ const LnNoAmountInvoicePaymentInput = new GT.Input({
   }),
 })
 
-const LnNoAmountInvoicePaymentSendMutation = GT.Field({
+const LnNoAmountInvoicePaymentSendMutation = GT.Field<
+  {
+    input: {
+      walletId: WalletId | InputValidationError
+      paymentRequest: EncodedPaymentRequest | InputValidationError
+      amount: Satoshis | InputValidationError
+      memo?: string | InputValidationError
+    }
+  },
+  null,
+  GraphQLContextForUser
+>({
   type: GT.NonNull(PaymentSendPayload),
   args: {
     input: { type: GT.NonNull(LnNoAmountInvoicePaymentInput) },
   },
   resolve: async (_, args, { domainAccount, logger }) => {
     const { walletId, paymentRequest, amount, memo } = args.input
-    for (const input of [walletId, memo, amount, paymentRequest]) {
-      if (input instanceof Error) {
-        return { errors: [{ message: input.message }] }
-      }
+
+    if (walletId instanceof InputValidationError) {
+      return { errors: [{ message: walletId.message }] }
+    }
+    if (paymentRequest instanceof InputValidationError) {
+      return { errors: [{ message: paymentRequest.message }] }
+    }
+    if (amount instanceof InputValidationError) {
+      return { errors: [{ message: amount.message }] }
+    }
+    if (memo instanceof InputValidationError) {
+      return { errors: [{ message: memo.message }] }
```

Comment:
why do we need to remove the  for? we are not doing additional logic

Reply:
typescript doesn't seem smart enough to handle the type from the loop?

Reply:
![image](https://user-images.githubusercontent.com/11156370/151141597-68dd7548-0dff-4c14-8c10-fad019d28d46.png)

Reply:
Yep